jsonschema/tests/test_validators.py

```diff
@@ -880,6 +880,17 @@ def test_invalid_properties(self):
         with self.assertRaises(SchemaError):
             self.Validator.check_schema({"properties": {"test": object()}})
 
+    def test_invalid_pattern(self):
+        with self.assertRaises(SchemaError):
+            self.Validator.check_schema({"pattern": "\q"})
+
+    @unittest.skip("This test fails but it shouldn't, see"
```

Comment:
Right so here, this test doesn't fail on draft 7, which properly says that the keys need to be regexes for `patternProperties`. It does that though using a new validator (`propertyNames`) that didn't exist before draft 6.

Draft 6 though seems to have a bug in the metaschema (by not specifying propertyNames for patternProperties even though that existed for draft 6), which sounds like something that should be filed  upstream.

And drafts 3 and 4 don't have that, so that's why this fails -- so it seems like the right thing to do isn't to skip this here, it should pass on draft 7, and draft 6 when/if the metaschema is fixed.

Reply:
Ah, right, I think the reason it's not there in draft6 is because draft6 (and 4) accidentally removed the `regex` format.

So, this seems to be a thing that really only should pass on draft 7, and "correctly" (FSVO correctly) blows up on earlier drafts.

Reply:
Understood, how can I skip the test on particular drafts? I don't understand the skip lambda https://github.com/Julian/jsonschema/blob/master/jsonschema/tests/test_jsonschema_test_suite.py#L81